contracts/lido_terra_hub/src/contract.rs

```diff
@@ -192,6 +171,80 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
     }
 }
 
+pub fn execute_claim_airdrops(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    token: String,
+    stage: u8,
+    amount: Uint128,
+    proof: Vec<String>,
+) -> StdResult<Response> {
+    let params: Parameters = PARAMETERS.load(deps.storage)?;
+    if params.paused.unwrap_or(false) {
+        return Err(StdError::generic_err("the contract is temporarily paused"));
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let owner = deps.api.addr_humanize(&config.creator)?;
+
+    if info.sender != owner {
+        return Err(StdError::generic_err("unauthorized"));
+    }
+
+    let withdrawal_account = deps.api.addr_humanize(
+        &config
+            .airdrop_withdrawal_account
+            .ok_or_else(|| StdError::generic_err("no withdrawal account configured"))?,
+    )?;
+
+    let registry_addr = deps
+        .api
+        .addr_humanize(&config.airdrop_registry_contract.ok_or_else(|| {
+            StdError::generic_err("the airdrop registry contract must have been registered")
+        })?)?;
+
+    let airdrop_info_resp: AirdropInfoResponse =
+        deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: registry_addr.to_string(),
+            msg: to_binary(&QueryAirdropRegistry::AirdropInfo {
+                airdrop_token: Some(token.clone()),
+                start_after: None,
+                limit: None,
+            })?,
+        }))?;
+
+    if airdrop_info_resp.airdrop_info.is_empty() {
+        return Err(StdError::generic_err(format!(
+            "no airdrop contracts found in the registry for token {}",
+            token
+        )));
+    }
+
+    let airdrop_info = airdrop_info_resp.airdrop_info[0].info.clone();
```

Comment:
i actually have a problem with this. see, we wanted this airdrops PR to be as lightweight as possible; it's totally okay to pass the `airdrop_contract` and `airdrop_token_contract` values as part of the `ClaimAirdrops` if it reduces the changeset. so i'm not sure that we need to refactor the airdrops contract